src/blocks/mrc_event_handler.ts

```diff
@@ -434,9 +452,22 @@ function createRobotEventHandlerBlock(
 export function addMechanismEventHandlerBlocks(
     mechanismInRobot: storageModuleContent.MechanismInRobot,
     events: storageModuleContent.Event[],
+    eventHandlerBlocks: EventHandlerBlock[],
     contents: toolboxItems.ContentsType[]) {
+  // Collect the ids of events for which there is already an event handler.
+  const eventIds: string[] = [];
+  eventHandlerBlocks.forEach(eventHandlerBlock => {
+    eventIds.push(eventHandlerBlock.getEventId());
+  });
   events.forEach(event => {
-    contents.push(createMechanismEventHandlerBlock(mechanismInRobot, event));
+    if (eventIds.includes(event.eventId)) {
+      // If there is already an event handler for this event, put a button in the toolbox.
+      const text = 'when ' + mechanismInRobot.name + ' ' + event.name;
```

Comment:
This needs to go through internationalization both for when and so the RTL languages will work correctly